src/app/components/elements/Userpic.jsx

```diff
@@ -1,51 +1,43 @@
 import React, {Component, PropTypes} from 'react';
 import {connect} from 'react-redux';
 import shouldComponentUpdate from 'app/utils/shouldComponentUpdate';
-import proxifyImageUrl from 'app/utils/ProxifyUrl';
+import { imageProxy } from 'app/utils/ProxifyUrl';
 
 class Userpic extends Component {
-    static propTypes = {
-        account: PropTypes.string
-    }
 
     shouldComponentUpdate = shouldComponentUpdate(this, 'Userpic')
 
     render() {
-        const {json_metadata, width, height} = this.props
+        const {account, json_metadata} = this.props
         const hideIfDefault = this.props.hideIfDefault || false
 
-        let url = null;
 
         // try to extract image url from users metaData
         try {
             const md = JSON.parse(json_metadata);
-            if(/^(https?:)\/\//.test(md.profile.profile_image)) {
-                const size = (width && width > 48)? '320x320' : '120x120';
-                url = proxifyImageUrl(md.profile.profile_image, size);
-            }
-        } catch (e) {}
-        if(!url) {
-            if(hideIfDefault) {
+            if(!/^(https?:)\/\//.test(md.profile.profile_image) && hideIfDefault) {
```

Comment:
Why is this check still needed?

Reply:
The code indicated there are situations where the user does not have a profile image set, and we do not want to show the default image. - There is at least one instance in condenser where that path could have been triggered (I think it's the user's page)

Reply:
Hm, why are we doing this? I thought it was so we didn’t have to make user lookups in condenser for every profile pic. Now we just loose the ability to dynamically size them and double the amount of calls to steemd

Reply:
This is only used for user page headers -- the default avatar looks pretty bad on the blueish gradient.

@jnordberg - I was under the impression that wherever the avatar image is used does not require any extra calls to steemd. This is because get_state includes author json for all accounts in a discussion. (However, now the image_hoster will be making a lot of extra get_account calls.. has the response time been measured? We may find that it needs to cache results for several minutes...)

Reply:
@roadscape - caching of avatars will be a fantastic feature for imagehoster, and I think it's part of the intended roadmap; [definitely worth logging another issue](https://github.com/steemit/imagehoster/issues/38).